core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts

```diff
@@ -165,6 +172,30 @@ export class DotExperimentsConfigurationGoalSelectComponent implements OnInit, O
                 ])
             })
         });
+
+        this.form.get('primary.type').valueChanges.subscribe((value) => {
+            this.defineDefaultName(value);
+        });
+    }
+
+    /**
+     * If the user don't set a custom name set the default name for the goal
+     * @param {string} typeValue
+     * @memberOf DotExperimentsConfigurationGoalSelectComponent
+     */
+    private defineDefaultName(typeValue: string): void {
+        const nameControl = this.form.get('primary.name') as FormControl;
+        if (
+            nameControl.value === this.DEFAULT_NAME_LABEL ||
```

Comment:
If the user changes the input value manually, and clicks the options, do not change the value. It is ok in that way?

Reply:
correct. that is the idea.